WooCommerce/Classes/ViewRelated/Editor/AztecEditorViewController.swift

```diff
@@ -0,0 +1,863 @@
+        var listIcons = [String: UIImage]()
+        for list in Constants.lists {
+            listIcons[list.formattingIdentifier.rawValue] = list.iconImage
+        }
+
+        listButton.alternativeIcons = listIcons
+
+        return [
+            headerButton,
+            listButton,
+            makeToolbarButton(identifier: .blockquote),
+            makeToolbarButton(identifier: .bold),
+            makeToolbarButton(identifier: .italic),
+            makeToolbarButton(identifier: .link)
+        ]
+    }
+
+    var overflowItemsForToolbar: [FormatBarItem] {
+        return [
+            makeToolbarButton(identifier: .underline),
+            makeToolbarButton(identifier: .strikethrough),
+            makeToolbarButton(identifier: .horizontalruler),
+            makeToolbarButton(identifier: .more),
+            makeToolbarButton(identifier: .sourcecode)
+        ]
+    }
+
+    // MARK: - Styling Options
+
+    private lazy var optionsTablePresenter = OptionsTablePresenter(presentingViewController: self, presentingTextView: editorView.richTextView)
+
+    /// Aztec's Awesomeness
+    ///
+    private var richTextView: Aztec.TextView {
+        get {
+            return editorView.richTextView
+        }
+    }
+
+    /// Raw HTML Editor
+    ///
+    private var htmlTextView: UITextView {
+        get {
+            return editorView.htmlTextView
+        }
+    }
+
+    /// Aztec's Text Placeholder
+    ///
+    fileprivate(set) lazy var placeholderLabel: UILabel = {
+        let label = UILabel()
+        label.text = NSLocalizedString("Start writing...", comment: "Aztec's Text Placeholder")
+        label.textColor = StyleManager.wooGreyMid
+        label.font = StyleManager.subheadlineFont
+        label.isUserInteractionEnabled = false
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.numberOfLines = 0
+        label.textAlignment = .natural
+        label.accessibilityIdentifier = "aztec-content-placeholder"
+        return label
+    }()
+
+    /// Current keyboard rect used to help size the inline media picker
+    ///
+    fileprivate var currentKeyboardFrame: CGRect = .zero
+
+    required init(content: String?) {
+        self.content = content ?? ""
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        registerAttachmentImageProviders()
+
+        configureNavigationBar()
+        configureView()
+        configureSubviews()
+
+        configureConstraints()
+
+        setHTML(content)
+
+        refreshPlaceholderVisibility()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        startListeningToNotifications()
+    }
+
+    deinit {
+        stopListeningToNotifications()
+    }
+}
+
+private extension AztecEditorViewController {
```

Comment:
It seems like the contents of this extension could also be extracted from the view controller. Combined with the configuration of other labels, maybe it could be worth exploring having a AztecUIConfigurator, only with a better name?